### PR TITLE
Bronwyn frequency encoder

### LINKED ISSUE #24: Code frequency_encoder
This is just an issue to track the progress of the entire function (since the milestone instructions state to create an issue for each function, and we only created issues for each docstring in particular).

- [x] Docstring
- [x] Code
- [x] Tests

Will modify as we understand what we need to do in the coming weeks.

### LINKED ISSUE #31: Code test_frequency_encoder
This is just an issue to track the progress of the entire unit test function for  #24.

- [x] Comments explaining each test.
- [x] 2 to 3 edge case tests

### PR DESCRIPTION
**Pull request by bbaillie**

### CODE REVIEW

File: encoderpy/frequency_encoder.py [hunk 1]
```diff
@@ -1,34 +1,63 @@
...
-        replaced by their encodings.
-  test_processed : pd.DataFrame
-        The test set, with the categorical columns specified by the argument cat_columns
-        replaced by the learned encodings from the training set.
+        Parameters
+        ----------
+        X_train : pd.DataFrame
+                A pandas dataframe representing the training data set containing some categorical features/columns.
+        X_test : pd.DataFrame
+                A pandas dataframe representing the test set, containing some set of categorical features/columns.
```
Comment (braydentang1) [inline encoderpy/frequency_encoder.py]: Maybe specify that this by default is None?

File: encoderpy/frequency_encoder.py [hunk 2]
```diff
@@ -1,34 +1,63 @@
...
+                in the test set. A larger value gives less weight to what is observed in the training set. A value
+                of 0 incorporates no prior information. The default value is 0.5.
+        Returns
+        -------
+        train_processed : pd.DataFrame
+                The training set, with the categorical columns specified by the argument cat_columns
+                replaced by their encodings.
+        test_processed : pd.DataFrame
+                The test set, with the categorical columns specified by the argument cat_columns
+                replaced by the learned encodings from the training set.
```
Comment (braydentang1) [inline encoderpy/frequency_encoder.py]: Maybe specify that this is returned if the user specifies a X_test only?

File: tests/test_frequency_encoder.py [hunk 3]
```diff
@@ -0,0 +1,32 @@
...
+    assert np.isnan(test_encode3['feature_cat_chr'].iloc[0]) == False,'The encoded value of an unseen class in test dataset is wrong'
+
+    #check shape
+    assert train_encode1.shape == train1.shape, "The shape of the encoded training dataset is wrong"
+    assert test_encode1.shape == test1.shape, "The shape of the encoded testing datset is wrong"
+
+    #check when X_test is none
+    assert train_encode2.shape == train1.shape, "The shape of the encoded training dataset is wrong when no X_test was provided"
+
+test_output()
```
Comment (braydentang1) [inline tests/test_frequency_encoder.py:32]: We will add more to this later so I think the tests are fine for now!

**Review by braydentang1 — CHANGES_REQUESTED** [review]

File: encoderpy/frequency_encoder.py (hunk 1 above)
Comment (bbaillie) [reply encoderpy/frequency_encoder.py]: just made this change

File: encoderpy/frequency_encoder.py (hunk 2 above)
Comment (bbaillie) [reply encoderpy/frequency_encoder.py]: will do

File: tests/test_frequency_encoder.py (hunk 3 above)
Comment (bbaillie) [reply tests/test_frequency_encoder.py:32]: awesome thanks

File: encoderpy/frequency_encoder.py (hunk 2 above)
Comment (bbaillie) [reply encoderpy/frequency_encoder.py]: done

File: encoderpy/frequency_encoder.py [hunk 4]
```diff
@@ -1,34 +1,75 @@
...
-        replaced by their encodings.
-  test_processed : pd.DataFrame
-        The test set, with the categorical columns specified by the argument cat_columns
-        replaced by the learned encodings from the training set.
+        Parameters
+        ----------
+        X_train : pd.DataFrame
+                A pandas dataframe representing the training data set containing some categorical features/columns.
+        X_test : pd.DataFrame
+                A pandas dataframe representing the test set, containing some set of categorical features/columns.
```
Comment (braydentang1) [inline encoderpy/frequency_encoder.py]:
```suggestion
                An optional pandas dataframe representing the test set, containing some set of categorical features/columns. Default is None.
```

**Review by braydentang1 — CHANGES_REQUESTED** [review]

**Review by braydentang1 — APPROVED** [review]
Looks good, thanks!
